### PR TITLE
Unexport client.KV and client.Txn.

### PR DESCRIPTION
**Pull request by petermattis**
Removed client.TransactionOptions.

### CODE REVIEW

**Comment by tamird** [comment]
LGTM

File: client/txn.go [hunk 1]
```diff
@@ -40,34 +40,34 @@ var (
...
 // access via Prepare and Flush. On receipt of
 // TransactionRestartError, the transaction epoch is incremented and
 // error passed to caller. On receipt of TransactionAbortedError, the
 // transaction is re-created and the error passed to caller.
 //
-// A Txn instance is not thread safe.
-type Txn struct {
-	kv           KV
+// A txn instance is not thread safe.
+type txn struct {
```
Comment (bdarnell) [inline client/txn.go]: So type `txn` contains a `proto.Transaction` also named `txn`, as well as a `txnSender` which embeds a `*txn` (and we also have a type `Tx`). This is all pretty confusing (and it appears to be confusing the compiler as well, if I'm interpreting the `txn.` additions above correctly). Let's rename `txn.txn` to `txn.txnProto`. Can we move the `Send` method to type `txn` and eliminate the `txnSender` type?

Is there any hope of merging this with the exported `Tx` type? If not is there a better name to express the distinction than `Tx` versus `txn`? Maybe `kvTxn`? For that matter, I'd also rename `Tx` to `Txn` now that the name is available - we use "txn" frequently but don't use "tx" anywhere else.

**Comment by bdarnell** [comment]
LGTM. We need to do something about naming here but it can be a future commit.

File: client/txn.go (hunk 1 above)
Comment (petermattis) [inline client/txn.go]: Yeah, I know this is quite ugly at the moment. I'm planning to merge `txn` into the `Tx` struct and probably rename `Tx` to `Txn`. And as you note, there is lots of other cleanup that can be done here. More PRs are coming.

**Pull request merged**